include/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.hpp

```diff
@@ -0,0 +1,251 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_Flight_Maneuver__
+#define __OpenSpaceToolkit_Astrodynamics_Flight_Maneuver__
+
+#include <OpenSpaceToolkit/Core/Container/Array.hpp>
+#include <OpenSpaceToolkit/Core/Container/Pair.hpp>
+#include <OpenSpaceToolkit/Core/Type/Real.hpp>
+#include <OpenSpaceToolkit/Core/Type/Shared.hpp>
+#include <OpenSpaceToolkit/Core/Type/Size.hpp>
+#include <OpenSpaceToolkit/Core/Type/String.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator.hpp>
+#include <OpenSpaceToolkit/Mathematics/Object/Vector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Data/Scalar.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Duration.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Unit.hpp>
+#include <OpenSpaceToolkit/Physics/Unit/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Unit/Length.hpp>
+#include <OpenSpaceToolkit/Physics/Unit/Mass.hpp>
+#include <OpenSpaceToolkit/Physics/Unit/Time.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubset.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/CartesianVelocity.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+
+using ostk::core::container::Array;
+using ostk::core::container::Pair;
+using ostk::core::type::Shared;
+using ostk::core::type::Size;
+using ostk::core::type::String;
+using ostk::core::type::Real;
+
+using ostk::mathematics::object::MatrixXd;
+using ostk::mathematics::object::VectorXd;
+using ostk::mathematics::object::Vector3d;
+using ostk::mathematics::curvefitting::Interpolator;
+
+using ostk::physics::coordinate::Frame;
+using ostk::physics::data::Scalar;
+using ostk::physics::time::Duration;
+using ostk::physics::time::Instant;
+using ostk::physics::Unit;
+using ostk::physics::unit::Derived;
+using ostk::physics::unit::Length;
+using ostk::physics::unit::Mass;
+using ostk::physics::unit::Time;
+
+using TabulatedDynamics = ostk::astro::dynamics::Tabulated;
+using ostk::astro::flight::system::PropulsionSystem;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianVelocity;
+
+#define DEFAULT_MANEUVER_INTERPOLATION_TYPE Interpolator::Type::BarycentricRational
+
+/// @brief Stores information and metadata about a maneuver as performed by a spacecraft.
+class Maneuver
```

Comment:
IMO most of the `getters` here should be `accessors`, per our discussion on this thread: https://loftorbital.slack.com/archives/C03GCNLG8G5/p1704928993120699
With the python methods as `get_*`

Reply:
yeah, lets resolve this once and for all because I believe we should use getters instead of accessors